ops/model.py

```diff
@@ -230,6 +247,84 @@ def _get_unique(self, relation_name, relation_id=None):
             raise TooManyRelatedAppsError(relation_name, num_related, 1)
 
 
+class BindingMapping(Mapping):
+    """A map of binding names to lists of Binding instances."""
+
+    def __init__(self, binding_names, backend):
+        """
+        binding_names -- Relation endpoint names and names of extra-bindings.
+        backend -- An instance of a model backend implementation.
+        """
+        self._backend = backend
+        self._data = {binding_name: None for binding_name in binding_names}
+
+    def __contains__(self, key):
+        return key in self._data
+
+    def __len__(self):
+        return len(self._data)
+
+    def __iter__(self):
+        return iter(self._data)
+
+    def __getitem__(self, binding_name):
+        binding_list = self._data[binding_name]
+        if binding_list is None:
+            binding_list = self._data[binding_name] = []
+            # A relation id can be None which means a binding is not associated with a relation.
+            binding_list.append(Binding(binding_name, None, self._backend))
+            for relation_id in self._backend.relation_ids(binding_name):
+                binding_list.append(Binding(binding_name, relation_id, self._backend))
+        return binding_list
+
+    def _get_unique(self, binding_name, relation_id=None):
+        if not isinstance(relation_id, (int, type(None))):
+            raise ModelError(f'relation name {relation_id} must be int or None not {type(relation_id).__name__}')
+        for binding in self[binding_name]:
+            if binding.relation_id == relation_id:
+                return binding
```

Comment:
What if it's not unique?  This logic is significantly different from what we offer on get_relation.